core/src/Kokkos_MemoryPool.hpp

```diff
@@ -585,19 +585,6 @@ class MemoryPool {
               (uint64_t(sb_id) << m_sb_size_lg2)       // superblock memory
               + (uint64_t(result.first) << size_lg2);  // block memory
 
-#if 0
-  printf( "  MemoryPool(0x%lx) pointer(0x%lx) allocate(%lu) sb_id(%d) sb_state(0x%x) block_size(%d) block_capacity(%d) block_id(%d) block_claimed(%d)\n"
-        , (uintptr_t)m_sb_state_array
-        , (uintptr_t)p
-        , alloc_size
-        , sb_id
-        , sb_state 
-        , (1u << size_lg2)
-        , (1u << count_lg2)
-        , result.first 
-        , result.second );
-#endif
-
```

Comment:
These are here for debug purposes.  How do you know it is ok to remove.

Reply:
That's why I opened a pull request: to see what people think about removing it. My preference goes towards code that is never possibly compiled. If you want to debug something, you would write this type of code again but I don't think it's good practice to keep it in the code base like this.

Reply:
If we were to do something like:

```
const bool debug = false;
if (debug)
  printf(...);

```

That should optimize out w/o warnings (a simple test under gcc10 and clang11 confirms this with -O0), yet the code itself is always checked by the compiler, and it is clear to the reader what this code is for.

I'm with @masterleinad